src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -588,17 +594,17 @@ def set_cloud_subscription(cli_ctx, cloud_name, subscription):
             config.remove_option(cloud_name, 'subscription')
         except configparser.NoSectionError:
             pass
-    if not os.path.isdir(GLOBAL_CONFIG_DIR):
-        os.makedirs(GLOBAL_CONFIG_DIR)
-    with open(CLOUD_CONFIG_FILE, 'w') as configfile:
+    if not os.path.isdir(os.path.dirname(get_cloud_config_file(cli_ctx))):
+        os.makedirs(os.path.dirname(get_cloud_config_file(cli_ctx)))
```

Comment:
This folder has been created by https://github.com/Azure/azure-cli/blob/63121edc73fa103658b3184d3d9d0328f43f330a/src/azure-cli-core/azure/cli/core/__init__.py#L77-L78
I don't know if this function is called when `cli_ctx` does not exist.